File: web/src/app/service/student.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpParams} from "@angular/common/http";
import {Observable} from "rxjs";
import {Student} from "../entity/student";
import {SchoolImpl} from "../entity/school";
import {ResponseBody} from "../entity/response-body";

@Injectable({
  providedIn: 'root'
})
export class StudentService {
  private baseUrl = 'http://localhost:8080/Student';
  constructor(private httpClient: HttpClient) { }

  add(student: {school: SchoolImpl, name: string, username: string, sno: string}): Observable<ResponseBody> {
    return this.httpClient.post<ResponseBody>(`${this.baseUrl}/add`, student);
  }

  changeStatus(id: number, status: number): Observable<Student> {
    return this.httpClient.put<Student>(`${this.baseUrl}/changeStatus/${id}`, { status });
  }

  delete(id: number): Observable<ResponseBody> {
    return this.httpClient.delete<ResponseBody>(`${this.baseUrl}/delete/${id}`);
  }

  getAll(): Observable<Student[]> {
    return this.httpClient.get<Student[]>(`${this.baseUrl}/getAll`);
  }

  getStudentById(id: number): Observable<Student> {
    return this.httpClient.get<Student>(`${this.baseUrl}/getStudentById/${id}`);
  }

  resetPassword(id: number, newPassword: string): Observable<ResponseBody> {
    return this.httpClient.patch<ResponseBody>(`${this.baseUrl}/resetPassword/${id}`, newPassword);
  }

  search(schoolId: number | null, searchName: string, searchStudentSno: string): Observable<Student[]> {
    let params = new HttpParams();

    if (schoolId) {
      params = params.set('schoolId', schoolId);
    }

    if (searchName) {
      params = params.set('searchName', searchName);
    }

    if (searchStudentSno) {
      params = params.set('searchStudentSno', searchStudentSno);
    }

    return this.httpClient.get<Student[]>(`${this.baseUrl}/search`, { params });
  }

  updateStudent(student: any): Observable<ResponseBody> {
    return this.httpClient.patch<ResponseBody>(`${this.baseUrl}/update/${student.id}`, student);
  }
}

File: api/src/main/java/com/mengyunzhi/eduPlanner/service/StudentService.java
package com.mengyunzhi.eduPlanner.service;


import com.mengyunzhi.eduPlanner.dto.Response;
import com.mengyunzhi.eduPlanner.dto.StudentRequest;
import com.mengyunzhi.eduPlanner.entity.Student;

import java.util.List;

public interface StudentService {

    /**
     * 新增学生
     * @param studentRequest 从前台传过来的数据类型
     * @return
     */
    Response<Void> save(StudentRequest studentRequest);

    /**
     * 获取所有学生
     * @return
     */
    List<Student> getAll();

    /**
     * 根据 userId 获取学生
     * @param userId
     * @return Student
     */
    Student findByUserId(Long userId);

    Response<String> delete (Long id);

    Student changeStatus(Long id, Long status);

    Response<Student> resetPassword(Long id, String newPassword);

    List<Student> search(Long schoolId, String searchName, String searchStudentSno);

    Student getStudentById(Long id);

    Response<String> updateStudent(Long id, StudentRequest studentRequest);
}
